de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/ScriptHandler.java

```diff
@@ -40,28 +37,24 @@
    *        simulation
    * @param exec KNIME way of managing storage and information output during the current NodeModels
    *        execution
-   * @param LOGGER A {@link NodeLogger} that keeps track of the progress during the execu-tion of
+   * @param logger A {@link NodeLogger} that keeps track of the progress during the execu-tion of
    *        the node
    * @param internalSettings internal settings to store the image file of the plot
    * @param nodeSettings settings of the node containing the dimensions of the output im-age
    * @throws Exception
    */
-  public final void runSnippet(final FskPortObject fskObj,
-      final FskSimulation simulation,
-      final ExecutionContext exec,
-      NodeLogger LOGGER,
-      File imageFile,
-      List<JoinRelationAdvanced> joinRelationList,
-      String suffix) throws Exception {
-    
+  public final void runSnippet(final FskPortObject fskObj, final FskSimulation simulation,
+      final ExecutionContext exec, NodeLogger logger, File imageFile,
+      List<JoinRelationAdvanced> joinRelationList, String suffix) throws Exception {
+
     // Sets up working directory with resource files. This directory needs to be deleted.
     exec.setProgress(0.05, "Add resource files");
     Optional<Path> workingDirectory;
     if (fskObj.getEnvironmentManager().isPresent()) {
       workingDirectory = fskObj.getEnvironmentManager().get().getEnvironment();
     } else {
       workingDirectory = Optional.of(FileUtil.createTempDir("tempResourceFiles").toPath());
-   
+
     }
     setWorkingDirectory(workingDirectory.get(), exec);

```

Comment:
You need to check if the optional workingDirectory is present. Anyway, since workingDirectory is always defined it makes no sense to make it optional. It could be a Path.